runtime/sema/check_assignment.go

```diff
@@ -136,6 +136,9 @@ func (checker *Checker) accessedSelfMember(expression ast.Expression) *Member {
 		members = containerType.Members
 	case *TransactionType:
 		members = containerType.Members
+	case *ReferenceType:
+		// self can only be a reference type if the container is an attachment, which is a composite
```

Comment:
Is this guaranteed to always be true? Or should there be a switch to potentially return an UnreachableError?

Reply:
Yes, it is always the case.